crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -117,16 +118,13 @@ pub fn apply_state_transition<S: States>(world: &mut World) {
         next_state_resource.set_changed();
 
         let exited = mem::replace(&mut world.resource_mut::<State<S>>().0, entered.clone());
-        world.run_schedule(OnExit(exited.clone()));
 
-        let transition_schedule = OnTransition {
+        // Try to run the schedules if they exist.
+        let _ = world.try_run_schedule(OnExit(exited.clone()));
```

Comment:
@james7132 Would you like these to use `.ok()` as well?

Reply:
IMO almost all of these should be instead of using `let _ = ...`